### ISSUE TITLE
General formatting is broken. 

### ISSUE THREAD

**Issue by hiimsergey** [tgrosinger/advanced-tables-obsidian#173]
The automatic table formatting, the one and only feature of the plugin, doesn't work correctly.

**How to reproduce?**
1. Open a page in Obsidian and go to a new line.
2. Start creating a usual table by starting with a pipe symbol "|", a text and then continuing with a Tab.
3. After creating multiple columns, create a new line with Enter. Leave it blank.
4. Repeat step 3 until the formatting isn't working correctly.

**Expected Behaviour**
The plugin fills the new line with pipe signs corresponding to the amount of columns.

**[Demonstration](https://user-images.githubusercontent.com/91432388/152389577-e504c213-65a8-4248-a5d8-d41036843d6d.mp4)**

**Setup**
OS: Windows 10
Obsidian: v0.13.23
Advanced Tables Plugin: v0.17.2

**Additional**
- No other plugins were used in the recreation process
- I have tried both editors, the old one and Live Preview

**Comment by lowkeysam** [comment]
Having a similar issue, have tried uninstalling and reinstalling

**Comment by tgrosinger** [comment]
Can you please try creating that table again, but wait a moment longer before pressing enter? In my test it seems like this was occurring intermittently, and the closest thing to a cause I could find was pressing the enter key very rapidly after entering content in a cell.

I'm guessing there may be an issue with the plugin attempting to make changes to the table while the last changes are still saving.

**Comment by hiimsergey** [comment]
Yes, I tried it again and waited for a sec before pressing Enter. Indeed the formatting was good again. So apparently it is an issue with the plugin "speed".

**Comment by luckman212** [comment]
I think I'm hitting this too. I made a screen recording:

https://user-images.githubusercontent.com/1992842/154749191-6c7c0f32-5cb5-4e78-83ed-0d3c9b3c6fc9.mp4

This is from Obsidian 0.13.25 and AT 0.17.2
macOS 12.2.1

Anything I can do to help test/debug this?

**Comment by tgrosinger** [comment]
Thanks @luckman212. I have been able to reproduce the issue as well, I just haven't had a chance to put time into Obsidian development recently. This will be top of my list though when I get the opportunity (unless someone beats me to it 🙂)
